### DISCUSSION TITLE
ConnectionsToLinks function in co-accessible networks  with Cicero vignette

### DISCUSSION THREAD

**Discussion by pengxin2019** [stuart-lab/signac#935] [Q&A] [answered]
Hi Tim,
I am following your vignette here: https://satijalab.org/signac/articles/cicero.html

Here are my generated conns, ccans and links objects: 
```
# > head(conns), A third column should contain the co-accessibility scores.
# Peak1             Peak2 coaccess
# 1 1-10000487-10000841 1-9756605-9757134        0
# 2 1-10000487-10000841 1-9761378-9762238        0
# 3 1-10000487-10000841 1-9765446-9766266        0
# 4 1-10000487-10000841 1-9767710-9768302        0
# 5 1-10000487-10000841 1-9770505-9770826        0
# 6 1-10000487-10000841 1-9784557-9785255        0
# > dim(conns)
# [1] 556348      3

# > head(ccans)
# Peak CCAN
# 1-100018539-100019553 1-100018539-100019553    1
# 1-100058819-100059526 1-100058819-100059526    1
# 1-100077390-100078099 1-100077390-100078099    1
# 1-100088559-100089520 1-100088559-100089520    1
# 1-100092950-100093504 1-100092950-100093504    1
# 1-100107061-100109125 1-100107061-100109125    1
# dim(ccans)#only chorm 1
# [1] 4257    2

links <- ConnectionsToLinks(conns = conns, ccans = ccans)#556348
# [1] "Coaccessibility cutoff used: 0.25"
# GRanges object with 192232 ranges and 2 metadata columns:
#   seqnames              ranges strand |       score     group
# <Rle>           <IRanges>  <Rle> |   <numeric> <numeric>
#   [1]        1  99864106-100019046      * |  0.00662879         1
# [2]        1  99915174-100019046      * |  0.00985291         1
# [3]        1  99976322-100019046      * |  0.01296190         1
# [4]        1 100019046-100059173      * |  0.24199922         1
# [5]        1 100019046-100077745      * |  0.22267850         1
# ...      ...                 ...    ... .         ...       ...
# [192228]        1    9999196-10086116      * | 0.024400188        68
# [192229]        1    9999196-10107509      * | 0.140145796        69
# [192230]        1    9999196-10117212      * | 0.000483947        69
# [192231]        1    9999196-10192332      * | 0.062415465        69
# [192232]        1    9999196-10233412      * | 0.002375634        72
# -------
```
I understand that there are less `ccans` than `conns` since not all  conns were not assigned a CCAN. Thus `peaks` of `ccans` is a subset of `conns`.

But I just do not understand the `ranges` in links . I still do not understand how `ConnectionsToLinks` works though I did read the descriiption of this function.  Why the `ranges` in `links` are not the same as the peaks of `ccans` or `conns`?? what is their difference?  I am so confused about the relationship of they three

In addition, should the `region`  of `CoveragePlot`  come from  `links` or `ccans`? 

Thanks
Best
Penny

**Comment by pengxin2019** [comment]
Hi there, 
anybody had an opportunity to take this question?
Thanks

**Comment by timoast** [comment, marked as answer]
Hi @pengxin2019, it says in the description for the `ConnectionsToLinks()` function: "Convert the output of Cicero connections to a set of genomic ranges where the start and end coordinates of the range are the midpoints of the linked elements. Only elements on the same chromosome are included in the output."

So the input (from cicero) is pairs of genomic ranges, the output is one genome range for each pair, the start being the midpoint of the first region, the end being the midpoint of the second region.

> In addition, should the region of CoveragePlot come from links or ccans?

The `region` parameter in `CoveragePlot()` is the region of the genome you want to plot, you define it however you like
